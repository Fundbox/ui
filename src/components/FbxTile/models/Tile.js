export class Tile {
  constructor(name, imgNormal, imgHover, imgDisabled, hoverColor = '#2c3e50', isEnabled = true, isConnected = false, metadata = {}) {
    this.name = name
    this.imgNormal = imgNormal
    this.imgHover = imgHover
    this.imgDisabled = imgDisabled
    this.hoverColor = hoverColor
    this.isEnabled = isEnabled
    this.isConnected = isConnected
    this.metadata = metadata
  }
}
