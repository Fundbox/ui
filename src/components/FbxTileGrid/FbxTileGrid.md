# FbxTileGrid
A tile grid component. 
will render a grid based on your column count definition.
Provides callbacks for click.
Animations based on images in assets.

<br>
<br>

# The `Tile` Constructor
This component exposes a `Tile` class which you should use to construct the tiles.

You will need to pass three type of  `<image/>.src` valid strings to this `Tile` constructor. You must supply normal, hover and disabled images. 
```js
const tiles = [
    new Tile( 'WellsFargo', <IMAGE_NORMAL>, <IMAGE_HOVER>, <IMAGE_DISABLED>, <HOVER_COLOR>, <IS_ENABELD>, <IS_CONNECTED>, <METADATA>),
    new Tile( 'Chase', <IMAGE_NORMAL>, <IMAGE_HOVER>, <IMAGE_DISABLED>, <HOVER_COLOR>, <IS_ENABELD>, <IS_CONNECTED>, <METADATA>),
    new Tile( 'BankOfAmerica', <IMAGE_NORMAL>, <IMAGE_HOVER>, <IMAGE_DISABLED>, <HOVER_COLOR>, <IS_ENABELD>, <IS_CONNECTED>, <METADATA>),
  ...
]
```
