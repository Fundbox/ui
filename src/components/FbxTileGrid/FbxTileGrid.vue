<template>
  <div class="container">
    <div class="tile"
         :class="tileClasses(tile)"
         v-for="tile in tiles"
         :key="tile.name"
         @mouseenter="onHoverOverTile($event, tile)"
         @mouseleave="onHoverOverTile($event, tile)"
         :style="{ width: (100 / columns) + '%' }"
    >
      <div class="tile-inner" >
        <template v-if="typeof tile === 'object'">
          <img class="normal" :src="tile.imgNormal" :alt="tile.name" v-show="tile.isEnabled">
          <img class="hover" :src="tile.imgHover" :alt="tile.name">
          <img class="disabled" :src="tile.imgDisabled" :alt="tile.name" v-show="!tile.isEnabled">
        </template>
      </div>
    </div>
  </div>
</template>

<script>
import { range } from 'lodash'
export default {
  name: 'FbxTileGrid',
  props: {
    columns: {
      type: Number,
      default: 1,
    },
    tileData: {
      type: Array,
      required: true,
    },
    shouldShowPlaceholders: {
      type: Boolean,
      default: false,
    }
  },
  methods: {
    tileClasses(tile) {
      return {
        placeholderTile: typeof tile === 'number',
        disabled: !tile.isEnabled,
        connected: tile.isConnected,
      }
    },
    onHoverOverTile(hoverEvent, tile) {
      if (tile.isEnabled) {
        let tileInner = hoverEvent.target.children[0]
        if (hoverEvent.type === 'mouseenter') {
          tileInner.style.background = tile.hoverColor
        } else {
          tileInner.style.background = ''
        }
      }
    },
  },
  computed: {
    tiles() {
      if (this.shouldShowPlaceholders) {
        // Calculate how many placeholder tiles to add to the array.
        const numberOfTiles = this.tileData.length
        const overBy = numberOfTiles % this.columns
        const rest = overBy === 0 ? 0 : this.columns - overBy
        return this.tileData.concat(range(0, rest))
      } else {
        return this.tileData
      }
    }
  },
}
</script>

<style lang="scss" scoped>
  @import "../../styles/utils/color-palette";

  $horizontal-spacing: 10px;
  $vertical-spacing: 10px;

  .container {
    // Negate the padding on tiles.
    margin: 0 (-$horizontal-spacing) (-$vertical-spacing) 0;
    display: flex;
    flex-wrap: wrap;
  }

  .tile {
    padding: 0 $horizontal-spacing $vertical-spacing 0;
    position: relative;
    height: 100px;
    cursor: pointer;
    flex-shrink: 0;
    flex-grow: 0;

    &.placeholderTile {
      .tile-inner {
        border: 2px solid $white;
        background: transparent;
      }
    }

    &-inner {
      background: $white;
      width: 100%;
      height: 100%;
      transition: background-color 200ms ease-in-out;
      position:  relative;
    }

    img {
      transition: opacity 200ms ease-in-out;
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
    }

    .hover {
      opacity: 0;
    }

    &:hover {
      .hover {
        opacity: 1;
      }
      .normal {
        opacity: 0;
      }
    }

    &.disabled {
      cursor: not-allowed;
    }

    &.connected {
      .tile-inner {
        border: 2px solid $dark-green;

        &:after {
          content: "";
          position: absolute;
          top: 0;
          right: 0;
          display: block;
          width: 15px;
          height: 15px;
          background: url("../../assets/icon-check-white.svg") $dark-green no-repeat 65% 40%;
        }
      }
    }
  }
</style>
