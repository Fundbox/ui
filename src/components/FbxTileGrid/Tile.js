export class Tile {
  constructor(name, imgNormal, imgHover, imgDisabled, isEnabled = true, isConnected = false) {
    Object.defineProperties(this, {
      name: { get() { return name } },
      imgNormal: { get() { return imgNormal }  },
      imgHover: { get() { return imgHover } },
      imgDisabled: { get() { return imgDisabled } },
      isEnabled: { get() { return isEnabled } },
      isConnected: { get() { return isConnected } },
    })
  }
}
