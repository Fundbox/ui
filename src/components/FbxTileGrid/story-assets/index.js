export { default as img5Normal } from './5_colored.svg'
export { default as img5Hover } from './5_white.svg'
export { default as img5Disabled } from './5_greyed.svg'
export { default as img459Normal } from './459_colored.svg'
export { default as img459Hover } from './459_white.svg'
export { default as img459Disabled } from './459_greyed.svg'
export { default as img524Normal } from './524_colored.svg'
export { default as img524Hover } from './524_white.svg'
export { default as img524Disabled } from './524_greyed.svg'
export { default as img643Normal } from './643_colored.svg'
export { default as img643Hover } from './643_white.svg'
export { default as img643Disabled } from './643_greyed.svg'
export { default as img745Normal } from './745_colored.svg'
export { default as img745Hover } from './745_white.svg'
export { default as img745Disabled } from './745_greyed.svg'
export { default as img782Normal } from './782_colored.svg'
export { default as img782Hover } from './782_white.svg'
export { default as img782Disabled } from './782_greyed.svg'
export { default as img783Normal } from './783_colored.svg'
export { default as img783Hover } from './783_white.svg'
export { default as img783Disabled } from './783_greyed.svg'
export { default as img1603Normal } from './1603_colored.svg'
export { default as img1603Hover } from './1603_white.svg'
export { default as img1603Disabled } from './1603_greyed.svg'
export { default as img2162Normal } from './2162_colored.svg'
export { default as img2162Hover } from './2162_white.svg'
export { default as img2162Disabled } from './2162_greyed.svg'
