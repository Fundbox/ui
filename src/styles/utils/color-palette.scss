$extra-light-red: #fff5d4;
$light-red: #e8a23e;
$medium-red: #eb5d5b;
$dark-red: #d60200;

$extra-light-green: #d9f5f5;
$light-green: #9ee6d9;
$medium-green: #00cca9;
$dark-green: #00bd9c;

$light-blue: #deebf9;
$medium-blue: #2c3e50;

$extra-light-gray: #f7f8f8;
$light-gray: #f0f2f2;
$medium-gray: #e3e5e5;
$extra-dark-gray: #858585;

$white: #fff;
$black: #000;

:export {
  extraLightRed: $extra-light-red;
  lightRed: $light-red;
  mediumRed: $medium-red;
  darkRed: $dark-red;
  extraLightGreen: $extra-light-green;
  lightGreen: $light-green;
  mediumGreen: $medium-green;
  darkGreen: $dark-green;
  lightBlue: $light-blue;
  mediumBlue: $medium-blue;
  extraLightGray: $extra-light-gray;
  lightGray: $light-gray;
  mediumGray: $medium-gray;
  extraDarkGray: $extra-dark-gray;
  white: $white;
  black: $black;
}
