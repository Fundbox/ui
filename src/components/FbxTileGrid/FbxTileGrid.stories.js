import { Tile } from '../FbxTile/models/Tile'
import { storiesOf } from '@storybook/vue'
import { withInfo } from 'storybook-addon-vue-info'
import { number as numberKnob, boolean as booleanKnob, object as objectKnob } from '@storybook/addon-knobs'
import { action } from '@storybook/addon-actions'
import FbxTileGrid from './FbxTileGrid.vue'
import summary from './FbxTileGrid.md'
import * as images from './story-assets'

const stories = storiesOf('Components/TileGrid', module)
const withSummary = withInfo({ summary })

const tileData = [
  new Tile('WellsFargo', images.img5Normal, images.img5Hover, images.img5Disabled),
  new Tile('UnionBank', images.img459Normal, images.img459Hover, images.img459Disabled),
  new Tile('UsBank', images.img524Normal, images.img524Hover, images.img524Disabled),
  new Tile('Chase', images.img643Normal, images.img643Hover, images.img643Disabled),
  new Tile('FifthThirdBank', images.img745Normal, images.img745Hover, images.img745Disabled),
  new Tile('Huntington', images.img782Normal, images.img782Hover, images.img782Disabled),
  new Tile('Regions', images.img783Normal, images.img783Hover, images.img783Disabled),
  new Tile('CitiBank', images.img1603Normal, images.img1603Hover, images.img1603Disabled),
  new Tile('PncBank', images.img2162Normal, images.img2162Hover, images.img2162Disabled),
]

function defaultStory() {
  return {
    data() {
      const numberOfTilesKnob = numberKnob('Number of tiles', tileData.length)
      const tileDataKnobed = tileData
        .map(tile => objectKnob(tile.name, tile))
        .map(tile => {
          if (tile instanceof Tile) {
            return tile
          } else {
            return new Tile(tile.name, tile.imgNormal, tile.imgHover, tile.imgDisabled, tile.hoverColor, tile.isEnabled, tile.isConnected, tile.metadata)
          }
        })
      return {
        tileData: tileDataKnobed.slice(0, numberOfTilesKnob),
      }
    },
    components: { FbxTileGrid },
    template: `
      <div style="background: #f7f8f8; padding: 20px; resize: horizontal; overflow: auto; border: 2px solid #d8d8d8">
        <fbx-tile-grid :columns="${numberKnob('Columns', 3)}" 
                       :tile-data="tileData"
                       :should-show-placeholders="${booleanKnob('Show placeholders', false)}"
                       :on-click="onClick"
                       >
        </fbx-tile-grid>
      </div>
    `,
    methods: {
      onClick: action('Tile was clicked')
    }
  }
}

stories.add('default', withSummary(defaultStory))
