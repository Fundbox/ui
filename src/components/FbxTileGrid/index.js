import FbxTileGrid from './FbxTileGrid.vue'
import { registerComponents, vueUse } from '../../utils/plugins'

const components = {
  FbxTileGrid
}

const VuePlugin = {
  install(Vue) {
    registerComponents(Vue, components)
  }
}

vueUse(VuePlugin)

export default VuePlugin
export * from '../FbxTile/models/Tile'
