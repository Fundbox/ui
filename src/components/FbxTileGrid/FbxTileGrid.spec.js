import { shallowMount } from '@vue/test-utils'
import FbxTileGrid from './FbxTileGrid.vue'
import { Tile } from './Tile'
import * as utils from './../../utils'

const genMockTileData = () => [
  new Tile('WellsFargo', 'story-assets/5_colored.svg', 'story-assets/5_white.svg', 'story-assets/5_white.svg'),
  new Tile('UnionBank', 'story-assets/459_colored.svg', 'story-assets/459_white.svg', 'story-assets/459_white.svg'),
  new Tile('UsBank', 'story-assets/524_colored.svg', 'story-assets/524_white.svg', 'story-assets/524_white.svg'),
  new Tile('Chase', 'story-assets/643_colored.svg', 'story-assets/643_white.svg', 'story-assets/643_white.svg'),
  new Tile('FifthThirdBank', 'story-assets/745_colored.svg', 'story-assets/745_white.svg', 'story-assets/745_white.svg'),
  new Tile('Huntington', 'story-assets/782_colored.svg', 'story-assets/782_white.svg', 'story-assets/782_white.svg'),
  new Tile('Regions', 'story-assets/783_colored.svg', 'story-assets/783_white.svg', 'story-assets/783_white.svg'),
  new Tile('CitiBank', 'story-assets/1603_colored.svg', 'story-assets/1603_white.svg', 'story-assets/1603_white.svg'),
  new Tile('PncBank', 'story-assets/2162_colored.svg', 'story-assets/2162_white.svg', 'story-assets/2162_white.svg'),
]

const tileClass = '.tile-wrapper'
const getNthTile = (wrapper, n = 1) => wrapper.find(`${tileClass}:nth-of-type(${n})`)

describe('Components/FbxTileGrid', () => {
  beforeEach(() => {
    // We re-assign a new noop function on each run to reset the function count
    utils.noop = jest.fn()
  })

  describe('snapshots', () => {
    it('renders 3 columns correctly', () => {
      const wrapper = shallowMount(FbxTileGrid, {
        propsData: {
          tileData: genMockTileData(),
          columns: 3,
        }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('renders one column correctly', () => {
      const wrapper = shallowMount(FbxTileGrid, {
        propsData: {
          tileData: genMockTileData().slice(0, 3),
          columns: 1,
        }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('renders disabled tile correctly', () => {
      const wrapper = shallowMount(FbxTileGrid, {
        propsData: {
          tileData: genMockTileData().map((tile, i) => {
            tile.isEnabled = i !== 0
            return tile
          }),
          columns: 3,
        }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('renders connected tile correctly', () => {
      const wrapper = shallowMount(FbxTileGrid, {
        propsData: {
          tileData: genMockTileData().map((tile, i) => {
            tile.isConnected = i === 1
            return tile
          }),
          columns: 3,
        }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('renders hover correctly', () => {
      const wrapper = shallowMount(FbxTileGrid, {
        propsData: {
          tileData: genMockTileData().map((tile, i) => {
            switch (i) {
              case 0:
                tile.isEnabled = false
                break
              case 1:
                tile.hoverColor = 'red'
                break
            }
            return tile
          }),
          columns: 3,
        }
      })
      getNthTile(wrapper).trigger('mouseenter')
      expect(wrapper.html()).toMatchSnapshot()
      getNthTile(wrapper).trigger('mouseleave')
      expect(wrapper.html()).toMatchSnapshot()
      getNthTile(wrapper, 2).trigger('mouseenter')
      expect(wrapper.html()).toMatchSnapshot()
      getNthTile(wrapper, 2).trigger('mouseleave')
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('renders placeholders correctly', () => {
      const wrapper = shallowMount(FbxTileGrid, {
        propsData: {
          tileData: genMockTileData().slice(0, 8),
          columns: 3,
          shouldShowPlaceholders: true,
        }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('renders NO placeholders when exactly fit in column correctly', () => {
      const wrapper = shallowMount(FbxTileGrid, {
        propsData: {
          tileData: genMockTileData(),
          columns: 3,
          shouldShowPlaceholders: true,
        }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('@events', () => {
    it('trigger onClick callback', () => {
      const mockClick = jest.fn()
      const wrapper = shallowMount(FbxTileGrid, {
        propsData: {
          tileData: genMockTileData(),
          columns: 3,
          onClick: mockClick,
        }
      })

      wrapper.find(`${tileClass}:first-of-type`).trigger('click')
      expect(mockClick).toHaveBeenCalledTimes(1)
      expect(mockClick.mock.calls[0][0]).toBeInstanceOf(Tile)
    })

    it('Check default callback (no callback passed)', () => {
      const wrapper = shallowMount(FbxTileGrid, {
        propsData: {
          tileData: genMockTileData(),
          columns: 3,
        }
      })
      getNthTile(wrapper).trigger('click')
      expect(utils.noop).toHaveBeenCalledTimes(1)
    })
  })
})
