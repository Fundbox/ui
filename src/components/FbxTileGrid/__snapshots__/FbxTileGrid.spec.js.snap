// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/FbxTileGrid snapshots renders 3 columns correctly 1`] = `
<div class="container">
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
</div>
`;

exports[`Components/FbxTileGrid snapshots renders NO placeholders when exactly fit in column correctly 1`] = `
<div class="container">
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
</div>
`;

exports[`Components/FbxTileGrid snapshots renders connected tile correctly 1`] = `
<div class="container">
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
</div>
`;

exports[`Components/FbxTileGrid snapshots renders disabled tile correctly 1`] = `
<div class="container">
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
</div>
`;

exports[`Components/FbxTileGrid snapshots renders hover correctly 1`] = `
<div class="container">
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
</div>
`;

exports[`Components/FbxTileGrid snapshots renders hover correctly 2`] = `
<div class="container">
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
</div>
`;

exports[`Components/FbxTileGrid snapshots renders hover correctly 3`] = `
<div class="container">
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
</div>
`;

exports[`Components/FbxTileGrid snapshots renders hover correctly 4`] = `
<div class="container">
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
</div>
`;

exports[`Components/FbxTileGrid snapshots renders one column correctly 1`] = `
<div class="container">
  <div class="tile-wrapper" style="width: 100%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 100%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 100%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
</div>
`;

exports[`Components/FbxTileGrid snapshots renders placeholders correctly 1`] = `
<div class="container">
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
  <div class="tile-wrapper" style="width: 33.333333333333336%;">
    <fbx-tile-stub tile="[object Object]"></fbx-tile-stub>
  </div>
</div>
`;
